File: src/main/java/tech/kavi/cms/web/user/ListHandler.java
package tech.kavi.cms.web.user;

import io.vertx.core.http.HttpMethod;
import io.vertx.ext.web.RoutingContext;
import org.springframework.beans.factory.annotation.Autowired;
import tech.kavi.cms.mapper.UserMapper;
import tech.kavi.vs.mybatis.AsyncHelperKt;
import tech.kavi.vs.web.ControllerHandler;
import tech.kavi.vs.web.HandlerRequest;

@HandlerRequest(path = "/list", method = HttpMethod.GET)
public class ListHandler extends ControllerHandler {

    @Autowired
    private UserMapper userMapper;

    @Override
    public void handle(RoutingContext routingContext){
        AsyncHelperKt.singleAsync(userMapper::listUser).subscribe(it -> {
            routingContext.response().end(it.toString());
        }, e-> {
            routingContext.response().end(e.getMessage());
        });
    }
}

File: src/main/kotlin/tech/kavi/cms/module/api/user/ListHandler.kt
package tech.kavi.cms.module.api.user

import io.vertx.core.http.HttpMethod
import io.vertx.core.json.JsonObject
import io.vertx.ext.web.Route
import io.vertx.ext.web.RoutingContext
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import tech.kavi.cms.dao.UserDao
import tech.kavi.vs.mybatis.singleAsync
import tech.kavi.vs.web.ControllerHandler
import tech.kavi.vs.web.HandlerRequest

@HandlerRequest(path = "/list", method = HttpMethod.GET)
class ListHandler @Autowired constructor(
        @Qualifier("config") val config: JsonObject,
        val userDao: UserDao
) : ControllerHandler() {
    /**
     * 自定义路由参数
     * */
    override fun route(route: Route) {
        super.route(route)
        route.order(1)
    }

    override fun handle(routingContext: RoutingContext) {
        singleAsync(userDao::listUser).subscribe({
            routingContext.response().end(it.toString())
        }, {
            routingContext.response().end(it.message)
        })
    }
}